lib/rdoc/markup/pre_process.rb

```diff
@@ -187,6 +187,12 @@ def handle_directive prefix, directive, param, code_object = nil,
       include_file filename, prefix, encoding
     when 'main' then
       @options.main_page = param if @options.respond_to? :main_page
+      warn <<~MSG
+        The :main: directive is deprecated and will be removed in RDoc 7.
+
+        Use `--main=#{param}` to specify the initial page displayed.
```

Comment:
I think it's also worth mentioning `main_page` for the `.rdoc_options` file as well.